server/src/main/java/org/opensearch/index/merge/MergeStats.java

```diff
@@ -224,6 +234,14 @@ public ByteSizeValue getCurrentSize() {
         return new ByteSizeValue(currentSizeInBytes);
     }
 
+    public void addUnreferencedFileCleanUpStats(UnreferencedFileCleanUpStats cleanUpStats) {
+        this.unreferencedFileCleanUpStats.add(cleanUpStats);
```

Comment:
It seems to me we are adding each time we are fetching the stats. Is this because we create new instance of engine after failure? In that case, what happens to previous count?

Reply:
This add function is called for specific scenario like when shard is closed. In that case [we add](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/indices/IndicesService.java#L583) the previous count to the current count so previous cleanup counts are not lost.